miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/file/MiniAppFileChooser.kt

```diff
@@ -36,15 +40,81 @@ interface MiniAppFileChooser {
     ): Boolean
 }
 
+/**
+ * A class to provide the interfaces for getting and requesting camera permission.
+ */
+interface MiniAppCameraPermissionDispatcher {
+    /**
+     * Get camera permission from host app.
+     * You can also throw an [Exception] from this method.
+     */
+    fun getCameraPermission(
+        onSuccess: (isGranted: Boolean) -> Unit,
+        onError: (message: String) -> Unit
+    ) {
+        throw MiniAppSdkException(ErrorBridgeMessage.NO_IMPL)
+    }
+    /**
+     * Request camera permission from host app.
+     * You can also throw an [Exception] from this method.
+     */
+    fun requestCameraPermission(
+        miniAppPermissionType: MiniAppDevicePermissionType,
+        permissionRequestCallback: (isGranted: Boolean) -> Unit
+    ) {
+        throw DevicePermissionsNotImplementedException()
+    }
+}
+
 /**
  * The default file chooser of a miniapp.
  * @param requestCode of file choosing using an intent inside sdk, which will also be used
  * to retrieve the data by [Activity.onActivityResult] in the HostApp.
  **/
+@Suppress("LargeClass", "TooManyFunctions")
 class MiniAppFileChooserDefault(var requestCode: Int) : MiniAppFileChooser {
 
     internal var callback: ValueCallback<Array<Uri>>? = null
     internal var currentPhotoPath: String? = null
+    private lateinit var miniAppCameraPermissionDispatcher: MiniAppCameraPermissionDispatcher
+    private var context: Context? = null
+
+    /**
+     * Set implemented miniAppCameraPermissionDispatcher.
+     * Can use the default provided class from sdk [MiniAppCameraPermissionDispatcher].
+     **/
+    fun setCameraPermissionDispatcher(miniAppCameraPermissionDispatcher: MiniAppCameraPermissionDispatcher) {
```

Comment:
Is it possible to pass `MiniAppCameraPermissionDispatcher` in the constructor of `MiniAppFileChooserDefault` instead of having a separate method for setting it?